### DISCUSSION TITLE
Can we include conditions to create Lineage

### DISCUSSION THREAD

**Discussion by ArunSP23EY** [OpenLineage/OpenLineage#1411] [General]
Hi Everyone,

I'm curious to learn the ability to control/manage the data lineage capturing ability of openlineage from databricks and importing the same in Azure Purview.

Say for example, if i require the lineage of a particular process (ETL from a Databricks notebook) to be inserted for first time and updated once a week/month. can it be achieved using openlineage ?

Thanks,
Arun

**Comment by mobuchowski** [comment]
@ArunSP23EY To be more clear, you want that for some runs (somehow defined) lineage would not be send?

**Comment by pawel-big-lebowski** [comment]
@ArunSP23EY I assume you mean Spark integration as you mentioned databricks platform. 

For Spark integration, we already do have a filtering mechanism with at least two filters defined. You can find it here -> https://github.com/OpenLineage/OpenLineage/tree/main/integration/spark/shared/src/main/java/io/openlineage/spark/agent/filters

To add a new filter, one need to:
 * Create a class that implements `io.openlineage.spark.agent.filters.EventFilter`
 * Filtering will be done in a method:
 ```
boolean isDisabled(SparkListenerEvent event)
```
 * Include newly created class in `io.openlineage.spark.agent.filters.EventFilterUtils`
